lib/wanda/execution/server.ex

```diff
@@ -64,6 +70,16 @@ defmodule Wanda.Execution.Server do
     end
   end
 
+  @impl true
+  def handle_info(
+        :timeout,
+        %State{} = state
+      ) do
+    :ok = Messaging.publish("checks.execution", :timeout)
```

Comment:
it is not clear from description, but we are splitting the PR in two because the second one touches the business logic and it would have been huge. basically we will get every agent that didn't answer in time and automatically fail every check it was supposed to send reporting the timeout in the event

Reply:
My bad :sweat_smile: I updated the PR's description, we're splitting up the effort because it's tough to just handle it altogether as @fabriziosestito says. For now I just implemented this simple timeout message, more will follow

Reply:
Okey! Thanks